ngraph_creator/operations/src/Reduce_Max.cpp

```diff
@@ -0,0 +1,57 @@
+#include <Reduce_Max.hpp>
+#define LOG_TAG "Reduce_Max"
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+Reduce_Max::Reduce_Max(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Reduce_Max::validate() {
+    // check output type
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkOutputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+
+    // Check all input types
+    if (!checkInputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkInputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM))
+        return false;
+
+    if (!checkInputOperandType(1, (int32_t)OperandType::TENSOR_INT32)) return false;
```

Comment:
Wouldn't you need a constant check for index 1 ? Dynamic dimensions might cause issue.

Reply:
currently there is no VTS with axis as input, so havent added this check.